src/myst/interpreter/native_lib/string.cr

```diff
@@ -57,18 +57,76 @@ module Myst
       TInteger.new(this.value.size.to_i64)
     end
 
+    NativeLib.method :string_chars, TString do
+      TList.new(this.value.chars.map { |c| TString.new(c.to_s).as Value })
+    end
+
+    NativeLib.method :string_downcase, TString do
+      TString.new(this.value.downcase)
+    end
+
+    NativeLib.method :string_upcase, TString do
+      TString.new(this.value.upcase)
+    end
+
+    NativeLib.method :string_chomp, TString, other : TString? do
+      other && return TString.new(this.value.chomp(other.value))
+      TString.new(this.value.chomp)
+    end
+
+    NativeLib.method :string_strip, TString do
+      TString.new(this.value.strip)
+    end
+    
+    NativeLib.method :string_rstrip, TString do
+      TString.new(this.value.rstrip)
+    end
+    
+    NativeLib.method :string_lstrip, TString do
+      TString.new(this.value.lstrip)
+    end
+
+    NativeLib.method :string_includes?, TString, other : TString do
+      TBoolean.new(this.value.includes?(other.value))
+    end
+
+    NativeLib.method :string_at, TString, index : TInteger, length : TInteger? do
+      begin
+        unless(length)
+          TString.new(this.value[index.value].to_s)
+        else # Length is provided - a range is requested
+          TString.new(this.value[index.value, length.value])
+        end
+      rescue 
```

Comment:
I would recommend against using the rescue for control flow

Reply:
Very Well, and Why is that?

Reply:
I don't think its necessary and leads to harder to read code (and possibly performance penalty?).  We can determine if we need to return `TNil` or not without waiting for the statement to fail.

Reply:
https://softwareengineering.stackexchange.com/a/189225

Generally, exceptions aren't meant to be caught locally: they more often propagate out and then some end-user handles them.

In this case, the same could be written without exceptions by explicitly showing the conditions:

```crystal
idx = index.value
length = length.value

result = 
  if length
    # Explicitly check that `String#[start, count]` will not fail.
    if idx < this.value.size && length > idx
      TString.new(this.value[idx, length])
    end
  else
    # Use nil-checking to assert that `index.value` is valid.
    if char = this.value[index.value]?
      TString.new(char.to_s)
    end
  end

# If none of the above conditions are met, default to `TNil`.
result ||= TNil.new
```